manifests/config.pp

```diff
@@ -21,12 +14,12 @@
     owner  => $katello_devel::user,
     group  => $katello_devel::group,
     mode   => '0755',
-  } ->
-  file { "${katello_devel::deployment_dir}/foreman/config/settings.plugins.d/katello.yaml":
-    ensure  => file,
-    content => template('katello/katello.yaml.erb'),
-    owner   => $katello_devel::user,
-    group   => $katello_devel::group,
-    mode    => '0644',
   }
+
+  katello_devel::plugin { 'katello/katello': }
+
+  $extra_plugins.each |String $plugin| {
```

Comment:
Probably not many foreman plugins do, I only know of a couple.  Having the git checkouts with remotes right and gems loaded is a pretty good first step for now.